### PR TITLE
Use markdig for Markdown

### LINKED ISSUE #726: Markdown doesn't display code blocks.
The markdown parser we're using doesn't display code blocks properly. For example [this PR](https://github.com/aspnet/Mvc/pull/5097) is displayed like this:

image

### LINKED ISSUE #765: Markdown doesn't display checkboxes
Similar to #726 and code blocks, checkboxes aren't rendered in the details view.

capture

### LINKED ISSUE #869: Markdown doesn't display tables
- GitHub Extension for Visual Studio  version: 2.2.0.6
- Visual Studio version: 2015 update 3

Markdown doesn't display tables
image

### LINKED ISSUE #870: Markdown doesn't strip html comments
- GitHub Extension for Visual Studio  version: 2.2.0.6
- Visual Studio version: 2015 update 3

Markdown doesn't strip html comments the same way Github.com does.

image

### PR DESCRIPTION
**Pull request by grokys**
After submitting #1016 I looked into creating a markdown viewer based on [markdig](https://github.com/lunet-io/markdig). There were the beginnings of WPF markdown support at https://github.com/grokys/markdig-wpf to which I submitted [three](https://github.com/Kryptos-FR/markdig-wpf/pull/3) [pull](https://github.com/Kryptos-FR/markdig-wpf/pull/4) [requests](https://github.com/Kryptos-FR/markdig-wpf/pull/5) to get markdown support to where we needed it.

Issues addressed:
- [x] #726 Markdown doesn't display code blocks.
- [x] #765 Markdown doesn't display checkboxes
- [x] #869 Markdown doesn't display tables
- [x] #870 Markdown doesn't strip html comments

To demonstrate the improvements you can open https://github.com/grokys/PullRequestSandbox/pull/26. These are the results:

Before (`master`):

image

With fixes in #1016:

image

With this PR:

image

Downsides:

- ~~PRs to Markdig-wpf haven't yet been merged and there's not yet a nuget package for it, so another submodule :(~~
- ~~A package for markdig is available, but it's not signed. For the moment using StrongNameSigner to sign this, but I know that trying to use that has caused us problems in the past. In addition markdig is >=VS2017 so wouldn't be so easy to include as a submodule.~~

Update: Applied #1040 to remove Fody and now StrongNameSigner seems to be working without a problem, so submodules are no longer needed.

Depends on #1040

Fixes #726
Fixes #765
Fixes #869
Fixes #870

### CODE REVIEW

**Comment by shana** [comment]
> PRs to Markdig-wpf haven't yet been merged and there's not yet a nuget package for it, so another submodule :(

We have a myget feed for our own signed packages and we're also already putting packages in the `lib` directory, so we don't have to include it as a submodule 😄

**Comment by shana** [comment]
Correction: I have a myget feed that I used at some point for some packages, but at the moment all the custom nuget packages we depend on (SQLitePCL.raw, Rx, EntryExitDecorator) are included in the `lib` directory in the repo.

**Comment by grokys** [comment]
@paladique yes probably, however there's now a NuGet package for markdig-wpf with my changes. I'm hoping that with #1040 we can use StrongNameSigner and use that package directly.

**Comment by grokys** [comment]
In a3cbd9a5d2592d2cd36a2e6900dcfff5170026cd I use StrongNameSigner to remove the necessity of including the package as a submodule.

File: src/StrongNameSign.targets [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	<ItemGroup>
+		<StrongNameSignPackage Include="Markdig"/>
+		<StrongNameSignPackage Include="Markdig.Wpf"/>
+	</ItemGroup>
+
+	<Target Name="BeforeBuild">
+		<Message Text="Running StrongNameSigner on packages: @(StrongNameSignPackage)..." Importance="high"/>
+		<Exec ContinueOnError="false"
+			  Command="&quot;$(MSBuildThisFileDirectory)..\packages\Brutal.Dev.StrongNameSigner.2.1.3\build\StrongNameSigner.Console.exe&quot; -in &quot;@(StrongNameSignPackage->'$(MSBuildThisFileDirectory)..\packages\%(Identity).*', '|')&quot;"/>
+	</Target>
```
Comment (shana) [inline src/StrongNameSign.targets]: Maybe put this next to the other targets files in the src/common folder, so they're all in one place?

Comment (grokys) [inline src/StrongNameSign.targets]: Good point - I didn't see the other targets there.

**Pull request merged**